lib/utils.js

```diff
@@ -222,3 +224,197 @@ var parseJSON = function parseJSON(data) {
+ *    dddddddd-dddd-dddd-dddd-dddddddddddd
+ *  -or-
+ *    {dddddddd-dddd-dddd-dddd-dddddddddddd}
+ * -or-
+ *    (dddddddd-dddd-dddd-dddd-dddddddddddd)
+ * -Groups of 8, 4, and 4 digits, and a subset of eight groups of 2 digits, with each group prefixed by "0x" or "0X", and separated by commas.
+ *    {0xdddddddd, 0xdddd, 0xdddd,{0xdd,0xdd,0xdd,0xdd,0xdd,0xdd,0xdd,0xdd}}
+ *
+ * @param {string} value - String value to be verified
+ * @returns {boolean} True if the value is a valid GUID
+ */
+var isValidGUID = function isValidGUID(value) {
+  // remove all embedded whitespaces
+  value = value.replace(/\s/g, '');
+
+  if ( /^[0-9a-fA-F]{1,32}$/i.test(value)
+    || /^[0-9a-fA-F]{1,8}-[0-9a-fA-F]{1,4}-[0-9a-fA-F]{1,4}-[0-9a-fA-F]{1,4}-[0-9a-fA-F]{1,12}$/i.test(value)
+    || /^\{[0-9a-fA-F]{1,8}-[0-9a-fA-F]{1,4}-[0-9a-fA-F]{1,4}-[0-9a-fA-F]{1,4}-[0-9a-fA-F]{1,12}\}$/i.test(value)
+    || /^\([0-9a-fA-F]{1,8}-[0-9a-fA-F]{1,4}-[0-9a-fA-F]{1,4}-[0-9a-fA-F]{1,4}-[0-9a-fA-F]{1,12}\)$/i.test(value)
+    || /^\{0x[0-9a-fA-F]{1,8},0x[0-9a-fA-F]{1,4},0x[0-9a-fA-F]{1,4},\{0x[0-9a-fA-F]{1,2},0x[0-9a-fA-F]{1,2},0x[0-9a-fA-F]{1,2},0x[0-9a-fA-F]{1,2},0x[0-9a-fA-F]{1,2},0x[0-9a-fA-F]{1,2},0x[0-9a-fA-F]{1,2},0x[0-9a-fA-F]{1,2}\}\}$/i.test(value)){
+    return true;
+  }
+  return false;
+}
+
+exports.isValidGUID = isValidGUID;
+
+/**
+ * Helper method to build the request options for Shared Key authentication.
+ *
+ * @param {object} options - Options on the form:
+ * ```js
+ * {
+ *    accountId: '...',                 // Azure storage accountId (required)
+ *    accessKey: '...',                 // Decoded Azure shared accessKey (required)
+ *    agent: '...',                     // HTTP Agent to use  (required)
+ *    hostname: '...',                  // The service hostname (required)
+ *    hasCanonicalizedHeaders: false,   // Specify if the string to sign includes the canonicalized headers (optional)
+ *    queryParamsSupported: '...'       // List of query-string parameter supported in lexicographical order, used for
+ *                                      // construction of the canonicalized resource.
+ *    headersValueToSign: '...'         // A string that includes all the headers value separated by a new line
+ *                                      // that should be included in the string to sign. (for more details,
+ *                                      // see the documentation of every service) (required)
+ * }
+ * ```
+ * @param {string} method - HTTP verb in upper case, e.g. `GET`.
+ * @param {string} path - Path on service resource for storage account.
+ * @param {object} query - Query-string parameters.
+ * @param {object} header - Mapping from header key in lowercase to value.
+ *
+ * @return {Promise} A promise for an options object compatible with `https.request`.
+ */
+var buildRequestOptionsForAuthSharedKey = function buildAuth(options, method, path, query, headers) {
```

Comment:
Reason: a simple copy-paste :) of another already created function. I wanted to be consistent with all the other functions created in utils.js :)